src/vm/gcheaputilities.h

```diff
@@ -192,11 +195,96 @@ class GCHeapUtilities {
     }
 #endif // FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
 
-
 private:
     // This class should never be instantiated.
     GCHeapUtilities() = delete;
 };
 
+// When not using per-thread allocation contexts, we (the EE) need to take care that
+// no two threads are concurrently modifying the global allocation context. This lock
+// must be acquired before any sort of operations involving the global allocation context
+// can occur.
+//
+// This lock is acquired by all allocations when not using per-thread allocation contexts. 
+// It is acquired in two kinds of places:
+//   1) JIT_TrialAllocFastSP (and related assembly alloc helpers), which attempt to 
+//      acquire it but move into an alloc slow path if acquiring fails
+//      (but does not decrement the lock variable when doing so)
+//   2) Alloc and AllocAlign8 in gchelpers.cpp, which acquire the lock using
+//      the Acquire and Release methods below.
+class GlobalAllocLock {
```

Comment:
The point is this is not a GC utility :) It's only used by allocators which are separate from the GC. I just looked and gchelpers.h really just includes the higher level allocation APIs for EE to use (so it's included in common.h). Since this is purely internal to allocation helpers I would just have the code in gchelpers.cpp since no one uses it except gchelpers.cpp.